src/components/PostsWrap.tsx

```diff
@@ -6,19 +6,46 @@ import GridDisplay from './GridDisplay';
 import ListDisplay from './ListDisplay';
 import { fetchRecommendations } from '../lib/supabase';
 import useSupabaseAuth from '../hooks/useSupabaseAuth';
+import { useLocation, useSearchParams } from 'react-router-dom';
 
 interface PostWrapProps{
   textToSearch: string
 }
 
-const PostsWrap = ({ textToSearch } :PostWrapProps): JSX.Element => {
+const locationsHash: { [index: string] : string | undefined } = {
+  "/most-popular-repositories": "popular",
```

Comment:
Please use the filter name instead. 

```suggestion
  "/popular": "popular",
```